esmf_regrid/tests/unit/schemes/_common_regridder.py

```diff
@@ -0,0 +1,283 @@
+
+    # Set REGRIDDER to the desired regridder class.
+    REGRIDDER = NotImplemented
+    # Configurable for the Nearest regridder.
+    N_LONS_SRC = 6
+
+    def test_dim_switching(self):
+        """
+        Test dimension order agnosticism.
+
+        Checks that the regridder accepts a cube with dimensions in a different
+        order than the cube which initialised it. Checks that dimension order is
+        inherited from the cube in the calling function in both cases.
+        """
+        n_lons = 6
+        n_lats = 5
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+        tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+        regridder = self.REGRIDDER(src, tgt)
+        unswitched_result = regridder(src)
+
+        src_switched = src.copy()
+        src_switched.transpose()
+        switched_result = regridder(src_switched)
+
+        assert unswitched_result.coord(dimensions=(0,)).standard_name == "latitude"
+        assert unswitched_result.coord(dimensions=(1,)).standard_name == "longitude"
+        assert switched_result.coord(dimensions=(0,)).standard_name == "longitude"
+        assert switched_result.coord(dimensions=(1,)).standard_name == "latitude"
+
+    def test_differing_grids(self):
+        """Test erroring when called on a different grid to the init grid."""
+        n_lons = 6
+        n_lats = 5
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src_init = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+        tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+        n_lons_dif = 7
+        src_dif_coord = _grid_cube(
+            n_lons_dif, n_lats, lon_bounds, lat_bounds, circular=True
+        )
+        src_dif_circ = _grid_cube(
+            n_lons, n_lats, lon_bounds, lat_bounds, circular=False
+        )
+
+        regridder = self.REGRIDDER(src_init, tgt)
+
+        msg = "The given cube is not defined on the same source grid as this regridder."
+        with pytest.raises(ValueError, match=msg):
+            _ = regridder(src_dif_coord)
+        with pytest.raises(ValueError, match=msg):
+            _ = regridder(src_dif_circ)
+
+    def test_invalid_mdtol(self):
+        """Test erroring when mdtol is out of range."""
+        n_lons = 6
+        n_lats = 5
+        lon_bounds = (-180, 180)
+        lat_bounds = (-90, 90)
+        src = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+        tgt = _grid_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+
+        match = "Value for mdtol must be in range 0 - 1, got "
+        with pytest.raises(ValueError, match=match):
+            _ = self.REGRIDDER(src, tgt, mdtol=2)
+        with pytest.raises(ValueError, match=match):
+            _ = self.REGRIDDER(src, tgt, mdtol=-1)
+
+    def test_curvilinear_equivalence(self):
```

Comment:
Note that this test diverges in behaviour with #276 due to differing behavour of regridders with respect to bounds.